### ISSUE TITLE
 no-undef rule false positive

### ISSUE THREAD

**Issue by sauron0914** [eslint/eslint#13821]
**Tell us about your environment**

Environment Info:

Node version: v14.8.0
npm version: v6.14.7
Local ESLint version: v7.12.1 (Currently used)
Global ESLint version: Not found

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
 no-undef: 2
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export function params (obj) {
  const str = []
  for (const k in obj) {
    if (typeof obj[k] !== 'undefined') {
      str.push(`${k}=${encodeURIComponent(obj[k])}`)
    }
  }
  return str.join('&')
}

export function parseSearch (query) {
  return `?${params(query)}`
}
```

```bash

```

**What did you expect to happen?**

do not error

**What actually happened? Please include the actual, raw output from ESLint.**

error    'params' is not defined

**Are you willing to submit a pull request to fix this bug?**
n/
